pkg/destroy/aws/aws.go

```diff
@@ -520,14 +525,19 @@ func deleteEC2InternetGateway(client *ec2.EC2, id string, logger logrus.FieldLog
 
 	for _, gateway := range response.InternetGateways {
 		for _, vpc := range gateway.Attachments {
+			if vpc.VpcId == nil {
+				logger.WithField("gateway", id).Warn("gateway does not have a vpc id")
+				continue
+			}
+			vpcID := *vpc.VpcId
 			_, err := client.DetachInternetGateway(&ec2.DetachInternetGatewayInput{
-				InternetGatewayId: gateway.InternetGatewayId,
-				VpcId:             vpc.VpcId,
+				InternetGatewayId: aws.String(id),
```

Comment:
`aws.String` is useful for string literals, but when you have a variable, you can just use `&id`.

Reply:
That assumes that I trust that the callee is not going to modify the value of `id`. But, sure, I can take out these changes.

Reply:
> That assumes that I trust that the callee is not going to modify the value of `id`.

I'm comfortable with that, and [`aws.String` doesn't make copies anyway][1].

[1]: https://github.com/aws/aws-sdk-go/blob/0943d6551d19b0eeee0b3f342659b5f4e871f025/aws/convert_types.go#L6-L7